src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -368,6 +397,25 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
 
     mem_info->type = UCS_MEMORY_TYPE_CUDA;
 
+    /* Synchronize future DMA operations */
+#if HAVE_CUDA_FABRIC
+    if (is_vmm_allocation) {
+        cu_err = cuCtxSetFlags(CU_CTX_SYNC_MEMOPS);
+        if (cu_err != CUDA_SUCCESS) {
+            ucs_warn("cuCtxSetFlags(CU_CTX_SYNC_MEMOPS) for %p error: %s",
+                     address, uct_cuda_base_cu_get_error_string(cu_err));
+        }
+    } else
+#endif
+    {
+        cu_err = cuPointerSetAttribute(&value, CU_POINTER_ATTRIBUTE_SYNC_MEMOPS,
+                                       (CUdeviceptr)address);
```

Comment:
What does this change fix?

Reply:
This change moves setting sync_memops to the common query function along with the change to do the same for VMM memory as setting pointer attribute doesn't work for VMM.